clang/lib/CodeGen/CGExprComplex.cpp

```diff
@@ -289,7 +289,7 @@ class ComplexExprEmitter
                                         const BinOpInfo &Op);
 
   QualType GetHigherPrecisionFPType(QualType ElementType) {
-    const auto *CurrentBT = dyn_cast<BuiltinType>(ElementType);
+    const auto *CurrentBT = cast<BuiltinType>(ElementType);
```

Comment:
This looks fine and is consistent with other type assumptions elsewhere in the same source file.